pkg/apis/policy/v1alpha1/propagation_types.go

```diff
@@ -152,6 +152,21 @@ type PropagationSpec struct {
 	// +kubebuilder:validation:Enum=Abort;Overwrite
 	// +optional
 	ConflictResolution ConflictResolution `json:"conflictResolution,omitempty"`
+
+	// ActivationStrategy declares how creation/modification/deletion to the Policy affect the resources it matches.
+	//
+	// The value "Immediate" means the matched resource will respond to the creation/modification/deletion immediately.
+	//
+	// The value "Delay" means the matched resource will delay to respond to the creation/modification/deletion until
+	// the resource itself is modified (In 1:N scenario where one Policy manages many resource templates,
```

Comment:
1. After the cluster label is modified, it triggers a rescheduling. Will the scheduler apply the new PP or old PP?
2. When introducing new features for PP in the future, do we need to consider whether this feature applies to `Delay` or `Immediate`?
3. Is this `activationStrategy` field allowed to be modified?
4. When the `activationStrategy` set to `Delay`, different resources matched by the same PP may apply different versions of PP. How can users distinguish which version of PP is applied to current resources?  -  When scheduling results do not meet expectations, how can users determine whether the scheduling result of current resources is due to a mistake in writing PP or a bug in scheduler?

Reply:
1. what does `cluster label` mean?
2. Actually, the behavior `Immediate` and `Delay` should be exactly the same, the only diffience should be before judging `CreateOrUpdateResourceBinding` in `Reconcile` of ResourceTemplate, we add a step to judge whether `Reconcile` is triggled by Karmada itself and the current bound Policy is `Delay`, then the resource skip to `CreateOrUpdateResourceBinding`. (It's just that there are some small problems in our code implementation, for example, Karmada add a label named `propagationpolicy.karmada.io/name` on the resource, which is also considered as a resource change, but this is actually Karmada's behaviour, and it should not considered as a resource change.)
3. yes
4. The real policy is in ResourceBinding，PropagationPolicy can be interpreted as a preconfigured Policy. If it is `Immediate`, it will Immediately refresh to ResourceBinding; if it is `Delay`, it will delay to refresh to ResourceBinding util user modified the Resource.

Reply:
> what does cluster label mean?

https://github.com/karmada-io/karmada/blob/master/pkg/scheduler/event_handler.go#L206


> The real policy is in ResourceBinding，PropagationPolicy can be interpreted as a preconfigured Policy. If it is Immediate, it will Immediately refresh to ResourceBinding; if it is Delay, it will delay to refresh to ResourceBinding util user modified the Resource.


It might be different, not all PP information is in RB.

My core question is: PP has multiple versions, how do users know which version they are currently using?

Reply:
**The `placement` in Binding is real Policy, while `placement` in PropagationPolicy is actually just preconfigured Policy.**

When the matched resource sync the `placement` in preconfigured Policy (PropagationPolicy) to real Policy (Binding), it means Policy take efftect.

When does the Policy take efftect, or saying when the matched resourece do this sync, it makes sense that it can be configured, just by this flied.

Even though PP has multiple versions, resource will always see just one version. Besides, without any doubt, resoure will response to each version Policy's changement, the only difference is before syncing the `placement` from current Policy to Binding (`CreateOrUpdateResourceBinding`), we add a step to check whether the syncing should be done immediately.

If you want to see the real `placement`, just see it in ResourceBinding. 
* The `placement` in PropagationPolicy must be euqal to the `placement` in ResourceBinding in case of the PropagationPolicy's `ActivationPreference` is `Immediate `.
* The `placement` in PropagationPolicy may be different with the `placement` in ResourceBinding in case of the PropagationPolicy's `ActivationPreference` is `Lazy `.

Reply:
Notes: my PR has been updated, the added field has been renamed to `ActivationPreference`

Reply:
When `ActivationPreference` is `Lazy`, does it mean that the `placement` in PP is Lazy, `Failover`, `ConflictResolution` etc. are `Immediate`, right?

Reply:
> When `ActivationPreference` is `Lazy`, does it mean that the `placement` in PP is Lazy, `Failover`, `ConflictResolution` etc. are `Immediate`, right?

Wrong. But you this comment is very important！！！I should update my code explanation !

`Failover` and `ConflictResolution` is the same as `placement`, this flied both exist in Policy and Binding, should be judge by `ActivationPreference` whether sync it to binding immediately.

A critical sentence is `The impact of Policy changes on the bondage relationship between resources and Policy is always immediate`, so some filed like `ResourceSelector`、`priority`、`premetion` are only works for "the bondage relationship between resources and Policy", since it has been solved, it no need sync to bindding.

So, if a filed in Policy is not works for "the bondage relationship between resources and Policy" and it not exist in binding, we should add it to binding.

Reply:
>. So, if a filed in Policy is not works for "the bondage relationship between resources and Policy" and it not exist in binding, we should add it to binding.

+1 We can proceed this way first, try it out for a while.

In the end, I suggest adding a featureGate to control this feature because I believe we haven't fully mastered it yet.

Reply:
The existing plan is to add a global default value. If you configured the `ActivationPreference`, use the configured value; if you didn't configure it, use global default value.

Shall we add another featureGate in addition to this global default value?

Reply:
I can accept this way.

Reply:
any other comments?

Reply:
>  So, if a filed in Policy is not works for "the bondage relationship between resources and Policy" and it not exist in binding, we should add it to binding.

I suggest that you submit the API modifications in a separate PR, which would make it more convenient for everyone to review.

Reply:
This PR is being discussed by some other partners, and it is not convenient for me to move it right away.
I will strictly abide by your this standard next time (ಥ_ಥ) .